### PR TITLE
fix: [CDS-74504]: modifies the call to get default values passed to response

### PR DESCRIPTION
**Pull request by sandeshsalunkhegh**
fix: [CDS-74504]: modifies the call for `pipelineExecutionSummaryEntity` creation before the pipeline execution to get default values passed from template to the response

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-74504]: https://harness.atlassian.net/browse/CDS-74504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)

### CODE REVIEW

File: 884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 public class InputSetTemplateHelper {
   // TODO(shalini): remove older methods with yaml string once all are moved to jsonNode
   public String createTemplateFromPipeline(String pipelineYaml) {
-    return RuntimeInputFormHelper.createRuntimeInputForm(pipelineYaml, true);
```
Comment (ankittiwari-harness) [inline 884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java:35]: Can you add tests

Comment (sandeshsalunkhegh) [reply 884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java:35]: @ankittiwari-harness Added tests.

**Comment by sandeshsalunkhegh** [comment]
Tested below possible cases:

Case 1: Both values, and defaults passed
Pipeline Input for Pipeline Execution:
Screenshot 2023-09-12 at 12 06 18 AM
Inputs:
Screenshot 2023-09-12 at 12 06 42 AM

Case 2: Values passed, defaults not passed
Pipeline Input for Pipeline Execution:
Screenshot 2023-09-12 at 12 04 52 AM
Inputs:
Screenshot 2023-09-12 at 12 05 18 AM

Case 3: Values not passed, and defaults passed
Pipeline Input for Pipeline Execution:
Screenshot 2023-09-12 at 12 03 08 AM
Inputs:
Screenshot 2023-09-12 at 12 03 43 AM

Case 4: Neither values, nor defaults passed
Pipeline Input for Pipeline Execution:
Screenshot 2023-09-11 at 11 12 50 PM
Inputs:
Screenshot 2023-09-11 at 11 12 07 PM

**Review by archit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6P31T7vKO9INHRo_X:-Ne6P31T7vKO9INHRo_Y:bzb0mm0) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L35)):*
> ```Java
>   // TODO(shalini): remove older methods with yaml string once all are moved to jsonNode
>   public String createTemplateFromPipeline(String pipelineYaml) {
>     return RuntimeInputFormHelper.createRuntimeInputFormWithDefaultValues(pipelineYaml);
> ```

this is one of the basic methods -> which caller needs this change?
___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6P6JB2-ECS0NNETK5:-Ne6P6JB2-ECS0NNETK6:bh9f3mt) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L38)):*
> ```Java
>   }
> 
>   public JsonNode createTemplateFromPipeline(JsonNode pipelineJsonNode) {
> ```

why we are not making change to this method?
cc: @shalini-agr 
___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6PA6J9zwN4AoSaudI:-Ne6PA6J9zwN4AoSaudJ:bqez30k) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L38)):*
> ```Java
>   }
> 
>   public JsonNode createTemplateFromPipeline(JsonNode pipelineJsonNode) {
> ```

Please add what is the change, what is the impact, what testing we did etc
cc: @brijeshdhakar @prashantsharma04

**Review by shalini-agr — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 3 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 51 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6_y4aB3E4n1q6au3v:-Ne6_y4aB3E4n1q6au3w:b-p0ukf8) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L51)):*
> ```Java
>   }
> 
>   public String createTemplateFromPipelineForGivenStages(String pipelineYaml, List<String> stageIdentifiers) {
> ```

we also need to send keepDefault true here.
Also, please test the case of selective stage execution, if we are getting default fields in the response in that case.

**Review by sandeshsalunkhegh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 3 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6P31T7vKO9INHRo_X:-Ne6nFPu4wzeci1L58Nk:b3kbldx) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L35)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

this is one of the basic methods -> which caller needs this change?
</blockquote></details>

@archit-harness We needed to pass on the default values for the variables, hence passing them by using an alternate function which passes `keepDefaultValues` as true.
Also, this change will add default values to the yaml, and not remove anything, so we should be good with this change and should not make any impact on further operations.
___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6P6JB2-ECS0NNETK5:-Ne6nbX11cdBHywRjHqQ:b-hyg32z) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L38)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why we are not making change to this method?
cc: @shalini-agr
</blockquote></details>

@archit-harness I didn't know these functions were related, I will work on this function and also add changes for this and the related functions. Thanks for pointing this out.
___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 38 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6PA6J9zwN4AoSaudI:-Ne6npZ2-J3fom9gVOty:b-4wgw50) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L38)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Please add what is the change, what is the impact, what testing we did etc
cc: @brijeshdhakar @prashantsharma04
</blockquote></details>

We needed to pass on the default values for the variables to the UI, hence passing them by using an alternate function which passes `keepDefaultValues` as true.
___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 51 at r3](https://reviewable.io/reviews/harness/harness-core/52715#-Ne6_y4aB3E4n1q6au3v:-Ne6nulR0r-yjjByFA41:bqrqrqg) ([raw file](https://github.com/harness/harness-core/blob/9912e6b699b90eb46fe1c5900d8ad584995abcd8/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L51)):*
<details><summary><i>Previously, shalini-agr (Shalini Agrawal) wrote…</i></summary><blockquote>

we also need to send keepDefault true here.
Also, please test the case of selective stage execution, if we are getting default fields in the response in that case.
</blockquote></details>

Yes @shalini-agr , I will add those in sometime.

**Review by prashantsharma04 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 4 file(s) to review. 2 change requests. 6. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 2 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 64 at r4](https://reviewable.io/reviews/harness/harness-core/52715#-Ne7TSDj9H-yhMXK-rbr:-Ne7TSDj9H-yhMXK-rbs:b3c8z94) ([raw file](https://github.com/harness/harness-core/blob/b0aa2702a1542dfee1bbade5f7524ca1ad76d424/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L64)):*
> ```Java
>       return null;
>     }
>     return removeNonRequiredStages(template, pipelineJsonNode, stageIdentifiers, false);
> ```

@sandeshsalunkhegh is there any case where the field is passed as false? if not, can we simply the method?

**Review by sandeshsalunkhegh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 2 change requests. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 64 at r4](https://reviewable.io/reviews/harness/harness-core/52715#-Ne7TSDj9H-yhMXK-rbr:-Ne8-OIl84ztmqhgjqfN:bpu8jdn) ([raw file](https://github.com/harness/harness-core/blob/b0aa2702a1542dfee1bbade5f7524ca1ad76d424/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L64)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

+1
</blockquote></details>

No, all the prior functions will send `true` for the argument variable `keepDefaultValues` to this function call.

**Review by shalini-agr — APPROVED** [review]

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 5 of 7 files at r5, 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 7 file(s) to review. 1 change requests. 3. unresolved discussions.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 7 files at r5, 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 4 file(s) to review. 1 change requests. 3. unresolved discussions. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 56 at r7](https://reviewable.io/reviews/harness/harness-core/52715#-NeDXfwB8qi7ceGCWxJ2:-NeDXfwB8qi7ceGCWxJ3:b-oickjr) ([raw file](https://github.com/harness/harness-core/blob/ffbbc8fb9e8a3461f25b10a300ceb9862f0e268b/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L56)):*
> ```Java
>       return null;
>     }
>     return removeNonRequiredStages(template, pipelineYaml, stageIdentifiers, false);
> ```

its earlier was true, why did we change it to false?
___
*[`953-yaml-commons/src/main/java/io/harness/pms/merger/helpers/RuntimeInputFormHelper.java` line 81 at r5](https://reviewable.io/reviews/harness/harness-core/52715#-NeDYHBa3uCXU6CLE4jO:-NeDYHBa3uCXU6CLE4jP:b-8rxk4f) ([raw file](https://github.com/harness/harness-core/blob/4fdd52897c0839cc19573d8f3aa359f5b526cfe8/953-yaml-commons/src/main/java/io/harness/pms/merger/helpers/RuntimeInputFormHelper.java#L81)):*
> ```Java
>   private YamlConfig createRuntimeInputFormYamlConfig(String yaml, boolean keepInput) {
>     YamlConfig yamlConfig = new YamlConfig(yaml);
>     return createRuntimeInputFormYamlConfig(yamlConfig, keepInput, false);
> ```

i didnt get why we are making it false?

**Review by sandeshsalunkhegh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: 4 file(s) to review. 1 change requests. 5. unresolved discussions. 

___
*[`884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java` line 56 at r7](https://reviewable.io/reviews/harness/harness-core/52715#-NeDXfwB8qi7ceGCWxJ2:-NeDuynMBAs2TRE_jALZ:bogxuwv) ([raw file](https://github.com/harness/harness-core/blob/ffbbc8fb9e8a3461f25b10a300ceb9862f0e268b/884-pms-commons/src/main/java/io/harness/pms/merger/helpers/InputSetTemplateHelper.java#L56)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

its earlier was true, why did we change it to false?
</blockquote></details>

No @archit-harness it was false earlier as well, I just reverted my changes back to what was there earlier, and modified the logic on how to solve that issue.
___
*[`953-yaml-commons/src/main/java/io/harness/pms/merger/helpers/RuntimeInputFormHelper.java` line 81 at r5](https://reviewable.io/reviews/harness/harness-core/52715#-NeDYHBa3uCXU6CLE4jO:-NeDusff7eDLysWeZMPw:b8n1v32) ([raw file](https://github.com/harness/harness-core/blob/4fdd52897c0839cc19573d8f3aa359f5b526cfe8/953-yaml-commons/src/main/java/io/harness/pms/merger/helpers/RuntimeInputFormHelper.java#L81)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

i didnt get why we are making it false?
</blockquote></details>

We are not making it false, I just reverted the code back to what was there earlier, and modified the logic on how to solve that issue.

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 7 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52715)* status: Some of the checks are too old. 2. unresolved discussions.

**Comment by sandeshsalunkhegh** [comment]
trigger gitleaks

**Pull request merged**